packages/pug-lexer/index.js

```diff
@@ -64,17 +64,14 @@ Lexer.prototype = {
     if (!value) this.error('ASSERT_FAILED', message);
   },
 
-  isExpression: function (exp) {
-    return isExpression(exp, {
-      throw: true
-    });
+  parseExpression: function (exp) {
+    return babylon.parseExpression(exp);
   },
 
   assertExpression: function (exp, noThrow) {
     //this verifies that a JavaScript expression is valid
     try {
-      this.callLexerFunction('isExpression', exp);
-      return true;
+      return this.callLexerFunction('parseExpression', exp);
     } catch (ex) {
```

Comment:
We may need to update the error handling code. Specifically the comment saying:

> not coming from acorn

is definitely out of date now.

Reply:
i removed the message. The rest of the code is working.